internal/v4/api.go

```diff
@@ -866,6 +868,58 @@ func (h *ReqHandler) putMetaPerm(id *router.ResolvedURL, path string, val *json.
 	return nil
 }
 
+// GET id/meta/homepage
+// https://github.com/juju/charmstore/blob/v4/docs/API.md#get-idmetahomepage
+func (h *ReqHandler) metaHomePage(entity *mongodoc.BaseEntity, id *router.ResolvedURL, path string, flags url.Values, req *http.Request) (interface{}, error) {
+	return params.HomePage {
+		HomePage: entity.HomePage.String(),
+	}, nil
+}
+
+// PUT id/meta/homepage
+// https://github.com/juju/charmstore/blob/v4/docs/API.md#put-idmeta
+func (h *ReqHandler) putMetaHomePage(id *router.ResolvedURL, path string, val *json.RawMessage, updater *router.FieldUpdater, req *http.Request) error {
+	var homePage params.HomePage
+	if err := json.Unmarshal(*val, &homePage); err != nil {
+		return errgo.Mask(err)
+	}
+
+	homeURL, err := url.Parse(homePage.HomePage)
+	if err != nil {
+		return errgo.Mask(err)
+	}
+
+	updater.UpdateField("homepage", homeURL, nil)
+	updater.UpdateSearch()
+	return nil
+}
+
+// GET id/meta/bugsurl
+// https://github.com/juju/charmstore/blob/v4/docs/API.md#get-idmetabugsurl
+func (h *ReqHandler) metaBugsURL(entity *mongodoc.BaseEntity, id *router.ResolvedURL, path string, flags url.Values, req *http.Request) (interface{}, error) {
+	return params.BugsURL {
```

Comment:
Now that I see how similar these are, I wonder if it would be more consistent to call the endpoints home-url/bugs-url or home-page/bugs-page. Also, perhaps only introduce a new URLResponse{URL string} in params for both?
Fee free to disagree, since this requires a bit of rework and I guess you followed the specs.